### PR TITLE
Add Support to Enum Type in Resource Function Params

### PR DESCRIPTION
**Pull request by DimuthuMadushan**
## Purpose

Support Enum Type in GraphQL Resource Function Input Params.

## Examples
```
enum Color {
  RED,
  GREEN,
  BLUE
}

service graphql:Service on new graphql:Listener(4000) {
    isolated resource function get color(Color color) returns Color {
        return RED;
    }
} 
```
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java [hunk 1]
```diff
@@ -173,14 +174,22 @@ private void validateInputParamType(FunctionDefinitionNode functionDefinitionNod
     }
 
     private boolean hasInvalidInputParamType(ParameterSymbol inputTypeSymbol) {
-        return !hasPrimitiveType(inputTypeSymbol.typeDescriptor());
+        boolean hasInvalidParamType = true;
+        if (inputTypeSymbol.typeDescriptor().typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            if ((((TypeReferenceTypeSymbol) inputTypeSymbol.typeDescriptor()).definition()).kind() == SymbolKind.ENUM) {
+                hasInvalidParamType = false;
```
Comment (ThisaruGuruge) [inline graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java:180]: can't we return directly here?
```suggestion
                return false;
```

File: graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java [hunk 2]
```diff
@@ -173,14 +174,22 @@ private void validateInputParamType(FunctionDefinitionNode functionDefinitionNod
...
+            }
+        } else {
+            hasInvalidParamType = !hasPrimitiveType(inputTypeSymbol.typeDescriptor());
```
Comment (ThisaruGuruge) [inline graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java:183]: Can return directly heree too:
```suggestion
            return !hasPrimitiveType(inputTypeSymbol.typeDescriptor());
```

File: graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java [hunk 3]
```diff
@@ -173,14 +174,22 @@ private void validateInputParamType(FunctionDefinitionNode functionDefinitionNod
     }
 
     private boolean hasInvalidInputParamType(ParameterSymbol inputTypeSymbol) {
-        return !hasPrimitiveType(inputTypeSymbol.typeDescriptor());
+        boolean hasInvalidParamType = true;
```
Comment (ThisaruGuruge) [inline graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java:177]: This variable definition can be removed if we directly return `false` when we find an invalid input type.

File: graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java [hunk 4]
```diff
@@ -173,14 +174,22 @@ private void validateInputParamType(FunctionDefinitionNode functionDefinitionNod
...
+        }
+        return hasInvalidParamType;
     }
 
     private boolean hasInvalidReturnType(TypeSymbol returnTypeSymbol) {
         boolean hasInvalidReturnType = false;
         if (returnTypeSymbol.typeKind() == TypeDescKind.MAP ||
-                returnTypeSymbol.typeKind() == TypeDescKind.JSON ||
-                returnTypeSymbol.typeKind() == TypeDescKind.BYTE) {
+            returnTypeSymbol.typeKind() == TypeDescKind.JSON ||
```
Comment (ThisaruGuruge) [inline graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java:191]: Wrrong indentation?

**Comment by ThisaruGuruge** [comment]
> ## Checklist
> * [ ]  Linked to an issue
> * [ ]  Updated the changelog
> * [x]  Added tests

Shall we link the related issue and update the changelog as well?

File: graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java [hunk 5]
```diff
@@ -83,12 +83,13 @@ private void validateResourceAccessorName(FunctionDefinitionNode functionDefinit
             if (resourceMethodSymbol.getName().isPresent()) {
                 if (!resourceMethodSymbol.getName().get().equals(PluginConstants.RESOURCE_FUNCTION_GET)) {
```
Comment (ayeshLK) [inline graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java:84]: * Could use following style instead.

```suggestion
            Optional<String> methodName = resourceMethodSymbol.getName();
            if (methodName.isPresent()) {
                if (!methodName.get().equals(PluginConstants.RESOURCE_FUNCTION_GET)) {
```

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Review by ayeshLK — APPROVED** [review]

**Pull request merged**